### PR TITLE
Bug 1156625 - [Messages] ThreadList&Thread separation: separate JS files

### PR DESCRIPTION
**Pull request by azasypkin**
At the current stage I move Thread/Composer/Report js files to views/thread/js - it seems that it will make next Thread&Composer separation easier - since all needed js files are already split from common and thread-list.

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2b75481df925ab0a7406a323685ec5060d0a5e6c

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9e8a93ab08eab1a93499579a0562edc9972f4c13

**Comment by julienw** [comment]
you should move the unit tests as well, otherwise they're not automatically run when you save the app file.

**Comment by julienw** [comment]
I'm unsure about information.js; maybe it should move to thread/ as well?

information should really be split in 2 so maybe this can wait for this.

**Comment by azasypkin** [comment]
> I'm unsure about information.js; maybe it should move to thread/ as well?
> information should really be split in 2 so maybe this can wait for this.

Yeah, I can move this to ```views/thread``` as I did for ```Composer``` part for now.

**Comment by azasypkin** [comment]
> you should move the unit tests as well, otherwise they're not automatically run when you save the app file.

Heh, didn't think about it, what do you think about mocks, should we move specific mocks to corresponding folders, @julienw ?

**Comment by azasypkin** [comment]
Now I think mocks should rather go to ```views/shared/test/unit``` as they used from different places

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=483865f984fa5084bb4ae3c8fe4767be13402e6f

**Comment by julienw** [comment]
> Now I think mocks should rather go to views/shared/test/unit as they used from different places

The path for unit tests is `/test/unit/views/...`

But otherwise I don't have big opinion for mocks.

**Comment by azasypkin** [comment]
> The path for unit tests is /test/unit/views/...

Mm, I tried this with ```views/thread/js/attachment.js``` if I change this file, then test isn't run automatically in neither of these folders:

* ```test/unit/views/attachment_test.js```
* ```test/unit/views/thread/attachment_test.js```
* ```test/unit/views/thread/js/attachment_test.js```

In log I see ```.... Running tests:  [ '/sms/views/thread/test/unit/attachment_test.js' ] ..... 0 passing (1ms)```, if I move test to ```views/thread/test/unit``` then I automatically re-runs when I change the file.

Am I missing something or I should re-configure it somewhere @julienw ?

**Comment by julienw** [comment]
When changing a file name (or adding a new file), you just need to restart the node server :)

Basically the path after /js should be the same than the path after /test/unit. If the path is /js/views/view.js, the unit test is /test/unit/views/view_test.js.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6acaa1543e02231d55c54f5cf299d1afaa931016

File: apps/sms/js/navigation.js [hunk 1]
```diff
@@ -42,17 +42,17 @@ var Navigation = {
...
-      behaviour: 'ThreadUI',
+      behaviour: 'ConversationView',
       wrapperPosition: 'left',
       container: 'thread-messages'
     },
     'thread-list': {
-      behaviour: 'ThreadListUI',
+      behaviour: 'InboxView',
       wrapperPosition: 'right',
       container: 'thread-list'
```
Comment (azasypkin) [inline apps/sms/js/navigation.js]: I didn't change names of the views in Navigation - I think it's a separate task.

File: apps/sms/views/conversation/test/unit/.jshintrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "../../../../test/unit/.jshintrc"
```
Comment (azasypkin) [inline apps/sms/views/conversation/test/unit/.jshintrc]: Once we create ```views/shared/test/unit```, this path should use shared one.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=723a0dcc7953b62277311d368da2dd962cda88e3

File: apps/sms/js/navigation.js (hunk 1 above)
Comment (steveck-chung) [inline apps/sms/js/navigation.js]: Ya it's better to change it together with html markup changes

**Comment by steveck-chung** [comment]
Seems the thread_list_mockup.js for MockThreadList still existed in unit test. It's fine if you want to create another bug for clean up.

**Comment by steveck-chung** [comment]
BTW some files' permission seems changed to 644, maybe you can set `git config core.fileMode false` to ignore the file mode changed, but I think 755/644 both works anyway.

**Comment by azasypkin** [comment]
> Seems the thread_list_mockup.js for MockThreadList still existed in unit test. It's fine if you want to create another bug for clean up.

I had mixed feelings about it, ```thread_list_mockup.js``` is more about array of threads rather then view

> BTW some files' permission seems changed to 644, maybe you can set git config core.fileMode false to ignore the file mode changed, but I think 755/644 both works anyway.

Hmmm interesting, let me check this, thanks!

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=73dfd601cca2ec6c21dcd3472c253f5243d42ece

**Pull request merged**